docs/pbc/all/payment-service-provider/202410.0/base-shop/payment-method-strategies.md

```diff
@@ -0,0 +1,131 @@
+
+## Express Checkout
+
+When the PaymentMethod is using the `express-checkout` strategy, specific buttons for this Payment Method are displayed in the shop. The buttons are displayed e.g. on the Cart Page or the PDP Page. The buttons are created by the PaymentMethodViewExpander. The PaymentMethodViewExpander is a part of the Payment module and is responsible for expanding the payment method data with the view data. The PaymentMethodViewExpander is used by the PaymentMethodViewPlugin. The PaymentMethodViewPlugin is a part of the Checkout module and is responsible for rendering the payment method data in the shop.
+
+To be able to display the buttons in the Shop, the CheckoutConfigurationTransfer contains all information needed to display the buttons. An example configuration looks as following:
+
+```json
+{
+   "base_url": "https://app-xyz.spryker.com",
+   "endpoints": [
+      ...
+   ],
+   "checkout_configuration": {
+      "strategy": "express-checkout",
+      "scripts": [
+         {
+            "url": "https://sandbox.paypal.com/sdk/js?client-id=AUn5n-...&merchant-id=3QK...&intent=authorize&commit=true&vault=false&disable-funding=card,sepa,bancontact&enable-funding=paylater",
+            "query_params": {
+               "currency": "currencyName",
+               "locale": "localeName"
+            }
+         }
+      ],
+      "payment_service_provider_data": {
+         "clientId": "AUn5n-...",
+         "merchantId": "3QK...",
+         "isLive": false
+      }
+   }
+}
+```
+
+The main elements of this JSON configuration are the strategy and the scripts.
+
+The `strategy` is set to `express-checkout` to define that the Payment Method should be displayed as a button in the shop.
+
+The `scripts` array contains the URLs to the scripts that are needed to display the button. The URL is used to load the script in the shop. The query_params are used to add key=value pairs to the URL with the actual values. The actual values are taken from the `\SprykerShop\Yves\PaymentAppWidget\Reader\PaymentMethodScriptReaderInterface` which is used in the Storefront to expand the script as needed. The `key` is the name that has to be used as query param key in the URL and the value defines which value should be used. The `currency` and `locale` are the most common values that are used in the scripts.
+
+The `payment_service_provider_data` contains the data that can be used by projects to configure the Payment Method. Those value may differ for each different Payment Method.
+
+### Query Params for Scripts
+
+It is important to understand who is responsible for what. The App itself knows only which query params (keys) are needed. The Shop knows which values are needed and where to get them from.  
+
+For example the script must be like `https://example.com/js?currency=EUR`. The App knows that the `currency` is needed but can't know which one is used in the Shop. The Shop knows that the used `currency` is `EUR`.
```

Comment:
The first sentence is not clear

Reply:
There is no better explanation for it.  

`App itself knows only which query params (keys)`
`The Shop knows which values`

we are talking about key=value pairs here

Reply:
I'm referring to the script and the URL. Not sure how they're related.

Reply:
Script URL and there is a typo Scrtipt.